File: app/src/main/java/app/kamy/saatApp/domain/model/VerseKeyFormat.kt
package app.kamy.saatApp.domain.model

object VerseKeyFormat {
    fun canonical(raw: String): String {
        if (!raw.startsWith("surah-")) return raw
        val rest = raw.removePrefix("surah-")
        val colon = rest.indexOf(':')
        if (colon < 0) return raw
        val chapterAyah = rest.substring(0, colon)
        val segments = chapterAyah.split('-')
        if (segments.size < 2) return raw
        val chapter = segments[0].toIntOrNull() ?: return raw
        val ayah = segments.last().toIntOrNull() ?: return raw
        return "$chapter:$ayah"
    }
}

File: iosApp/Saat/Domain/Enums/VerseKeyFormat.swift
//
//  VerseKeyFormat.swift
//  Saat
//
//  Created by Elmee on 25/04/2026.
//  Copyright © 2026 Elmee. All rights reserved.
//

import Foundation

enum VerseKeyFormat {
    static func canonical(from raw: String) -> String {
        if raw.hasPrefix("surah-") {
            let rest = String(raw.dropFirst("surah-".count))
            guard let colon = rest.firstIndex(of: ":") else { return raw }
            let chapterAyah = rest[..<colon]
            let segments = chapterAyah.split(separator: "-")
            if segments.count >= 2,
               let chapter = Int(segments[0]),
               let ayah = Int(segments[segments.count - 1]) {
                return "\(chapter):\(ayah)"
            }
        }
        return raw
    }

    @MainActor
    static func humanLabel(for verseKey: String) -> String {
        let key = canonical(from: verseKey)
        let parts = key.split(separator: ":")
        guard parts.count == 2,
              let chapter = Int(parts[0]),
              let ayah = Int(parts[1]) else {
            return verseKey
        }
        let surahName = ChapterCatalog.displayName(forChapterId: chapter) ?? "Surah \(chapter)"
        return "\(surahName)・\(ayah)"
    }
}
